community/detectors/geoserver_cve_2024_36401/src/test/java/com/google/tsunami/plugins/detectors/rce/GeoserverCve202436401VulnDetectorTest.java

```diff
@@ -0,0 +1,190 @@
+
+  TargetInfo targetInfo;
+  NetworkService targetNetworkService;
+  private final SecureRandom testSecureRandom =
+      new SecureRandom() {
+        @Override
+        public void nextBytes(byte[] bytes) {
+          Arrays.fill(bytes, (byte) 0xFF);
+        }
+      };
+
+  @Bind(lazy = true)
+  @OobSleepDuration
+  private int sleepDuration = 1;
+
+  @Before
+  public void setUp() throws IOException {
+    mockCallbackServer.start();
+    Guice.createInjector(
+            new FakeUtcClockModule(fakeUtcClock),
+            new HttpClientModule.Builder().build(),
+            FakePayloadGeneratorModule.builder()
+                .setCallbackServer(mockCallbackServer)
+                .setSecureRng(testSecureRandom)
+                .build(),
+            Modules.override(new GeoserverCve202436401VulnDetectorBootstrapModule())
+                .with(BoundFieldModule.of(this)))
+        .injectMembers(this);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    mockTargetService.shutdown();
+    mockCallbackServer.shutdown();
+  }
+
+  @Test
+  public void detect_whenVulnerable_returnsVulnerability() throws IOException {
+    startMockWebServer();
+    mockCallbackServer.enqueue(PayloadTestHelper.generateMockSuccessfulCallbackResponse());
+
+    DetectionReportList detectionReports =
+        detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+
+    assertThat(detectionReports.getDetectionReportsList())
+        .containsExactly(
+            DetectionReport.newBuilder()
+                .setTargetInfo(targetInfo)
+                .setNetworkService(targetNetworkService)
+                .setDetectionTimestamp(
+                    Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                    Vulnerability.newBuilder()
+                        .setMainId(
+                            VulnerabilityId.newBuilder()
+                                .setPublisher("TSUNAMI_COMMUNITY")
+                                .setValue("GeoserverCve202436401"))
+                        .setSeverity(Severity.CRITICAL)
+                        .setTitle("Geoserver RCE (CVE-2024-36401)")
+                        .setDescription(
+                            "This detector checks for Geoserver RCE (CVE-2024-36401). "
+                                + "Multiple OGC request parameters allow Remote Code Execution (RCE) "
+                                + "by unauthenticated users through specially crafted input against "
+                                + "a default GeoServer installation due to unsafely evaluating property "
+                                + "names as XPath expressions.")
+                        .setRecommendation(
+                            "Upgrade Geoserver to a patched version. The vulnerability was fixed in versions 2.23.6, 2.24.4, and 2.25.2.")
```

Comment:
```suggestion
                        .setDescription(
                            "This detector checks for Geoserver RCE (CVE-2024-36401). Multiple OGC"
                                + " request parameters allow Remote Code Execution (RCE) by"
                                + " unauthenticated users through specially crafted input against a"
                                + " default GeoServer installation due to unsafely evaluating"
                                + " property names as XPath expressions.")
                        .setRecommendation(
                            "Upgrade Geoserver to a patched version. The vulnerability was fixed in"
                                + " versions 2.23.6, 2.24.4, and 2.25.2.")
```
Split overlong strings